src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -44,100 +46,69 @@
-                    if (!userId.equals(userIdR)) {
-                        throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
-                    }
-                    String medicalRecord = null;
-                    List<String> oncHistoryIDs = null;
-                    if (StringUtils.isNotBlank(ddpParticipantId)) {
-                        logger.info(request.url());
-                        String pdfType = null;
-                        if (request.url().endsWith(CONSENT_PDF)) {
-                            pdfType = CONSENT_PDF;
-                        }
-                        else if (request.url().endsWith(RELEASE_PDF)) {
-                            pdfType = RELEASE_PDF;
-                        }
-                        else if (request.url().contains(COVER_PDF)) {
-                            pdfType = COVER_PDF;
-                            medicalRecord = request.params(RequestParameter.MEDICALRECORDID);
+                if (request.url().contains(RoutePath.DOWNLOAD_PDF)) {
+                    String requestBody = request.body();
+                    if (StringUtils.isNotBlank(requestBody)) {
+                        JSONObject jsonObject = new JSONObject(requestBody);
+                        String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
+                        Integer userIdRequest = Integer.parseInt(userIdR);
+                        if (!userId.equals(userIdR)) {
+                            throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
                         }
-                        else if (request.url().contains(REQUEST_PDF)) {
-                            pdfType = REQUEST_PDF;
-                            if (queryParams.value("requestId") != null) {
-                                oncHistoryIDs = Arrays.asList(queryParams.get("requestId").values());
+
+                        String ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
+                        if (StringUtils.isNotBlank(ddpParticipantId)) {
+                            String configName = null;
+                            if (jsonObject.has(RequestParameter.CONFIG_NAME)) {
+                                configName = (String) jsonObject.get(RequestParameter.CONFIG_NAME);
                             }
-                        }
-                        else if (request.url().endsWith(PDF) && StringUtils.isNotBlank(configName)) {
-                            pdfType = configName;
-                        }
-                        if (StringUtils.isNotBlank(pdfType)) {
-                            getPDFs(response, ddpParticipantId, pdfType, medicalRecord, oncHistoryIDs, realm, requestBody, userIdRequest);
+                            String medicalRecord = null;
+                            if (jsonObject.has("medicalRecordId")) {
```

Comment:
All this getting value logic of ifs from 87 to 98 can be wrapped in `PDFPayload` with using Gson's `fromJson`.